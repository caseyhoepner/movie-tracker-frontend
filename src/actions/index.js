export const addMovies = (movies) => ({
  type: 'ADD_MOVIES',
  movies
})

export const toggleFavorite = (movieId) => ({
  type: 'TOGGLE_FAVORITE',
  movieId: movieId
})

export const setUser = (userId) => ({
  type: 'SET_USER',
  userId: userId
})

export const removeUser = () => ({
  type: 'REMOVE_USER'
})

export const filterMovies = (movies) => ({
  type: 'FILTER_MOVIES',
  movies
})