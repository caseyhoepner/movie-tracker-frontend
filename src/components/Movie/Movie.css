.m-movie {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-shadow: 0px 7px 12px 0px #00000040;
  margin: 15px;
  padding: 0;
}

.m-poster {
  width: 250px;
}

.m-fav-container {
  font-family: Futura, sans-serif;
  font-size: 1.15rem;
  display: flex;
  align-items: center;
  justify-content: space-around;
  width: 100%;
  background-color: #D4A931;
  height: 40px;
  cursor: pointer;
  border: none;
}

.m-fav-text {
  color: #48788C;
}

.m-fav-btn {
  height: 30px;
  width: 30px;
  background-image: url('../../assets/images/hatOutline-blue.svg');
  background-size: cover;
  display: flex;
  align-items: center;
  justify-content: center;
  color: #ffffff;
}

.m-fav-btn:focus {
  outline: none;
}

.m-fav-btn.red{
  background-image: url('../../assets/images/hatFull-red.svg');
}