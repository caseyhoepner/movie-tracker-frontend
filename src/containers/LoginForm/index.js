import React, { Component } from 'react'

export class LoginForm extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  render() {
    return(
      <form>
        <input />
      </form>
    )
  }
}

export default LoginForm;