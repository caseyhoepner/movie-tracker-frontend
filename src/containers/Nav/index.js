import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { removeUser, filterMovies } from '../../actions';
import userIcon from '../../assets/images/margot.png';
import './Nav.css';

export const Nav = (props) => {
  const { userLoggedIn, removeUser, movies, filterMovies } = props

  const handleHover = (event) => {
    if(!event.target.classList.contains('active')) {
      event.target.classList.add('active')
    } else {
      event.target.classList.remove('active')
    }
  }

  return (
    <nav className="nav-container">
      <NavLink
        onClick={() => filterMovies(movies)}
        to="/favorites"
        className="fav-btn-group"
      >
        <div
          className="fav-btn-nav"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <span className="num-favs">0</span>
          <p
            className={`favorites-text`}
          >
            FAVORITES
          </p>
        </div>
      </NavLink>
      <NavLink
        to="/login"
        className={userLoggedIn ? "hidden" : "show"}
      >
        <div
          className="login-btn"
          onMouseOver={handleHover}
          onMouseOut={handleHover}
        >
          <img
            className="login-icon"
            src={ userIcon }
            alt='Click here to login'/>
          <p className="login-text">
            LOGIN
          </p>
        </div>
      </NavLink>
      <NavLink
        className={userLoggedIn ? "logout-btn" : "hidden"}
        onClick={removeUser}
        to='/'
      >
        <img
          className="login-icon"
          src={ userIcon }
          alt='Click here to login'/>
        <p className="login-text">
          LOGOUT
        </p>
      </NavLink>
    </nav>
  )
}

export const mapStateToProps = (state) => ({
  userLoggedIn: state.user.id,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  removeUser: () => dispatch(removeUser()),
  filterMovies: (movies) => dispatch(filterMovies(movies))
})

Nav.propTypes = {
  removeUser: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.number
}

export default connect(mapStateToProps, mapDispatchToProps)(Nav);