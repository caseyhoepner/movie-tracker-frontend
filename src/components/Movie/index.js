import React from 'react';
import PropTypes from 'prop-types';
import './Movie.css';
import fullHeart from '../../assets/images/full-heart.svg'
import emptyHeart from '../../assets/images/empty-heart.svg'


const Movie = ( { poster_path } ) => {

  return (
    <div className='movie'>
      <div className='movie-poster'>
        <img 
          className='poster'
          src={`http://image.tmdb.org/t/p/w185//${poster_path}`}
        />
      </div>
      <div className='fav-container'>
        <p className='fav-text'>FAVORITE</p>
        <div className='fav-btn'></div>
      </div>
    </div>
  )
}

export default Movie;