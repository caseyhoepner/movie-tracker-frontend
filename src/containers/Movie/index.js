import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';

import * as API from '../../utils'
import { toggleFavorite } from '../../actions';
import './Movie.css';

// make functional
export class Movie extends Component {
  handleClick = async () => {
    const { movie_id, overview, poster_path, release_date, title, vote_average, isLoggedIn, toggleFavorite } = this.props;
    const movie = {
      user_id: isLoggedIn,
      movie_id: movie_id,
      overview,
      poster_path,
      release_date,
      title,
      vote_average,
    }

    await API.addFavorite(movie)
    toggleFavorite(movie_id)
  }

  render() {
    let favoriteBtn;
    const { poster_path, movie_id, isLoggedIn, isFavorite } = this.props;

    if (isLoggedIn) {
      favoriteBtn =
        <button className='m-fav-container'
          onClick={this.handleClick}
        >
          <p className='m-fav-text'>FAVORITE</p>
          <div
            className={`m-fav-btn ${isFavorite ? 'red' : ''}`}>
          </div>
        </button>
    } else {
      favoriteBtn =
        <NavLink className='m-fav-container'
          to='/login'>
          <p className='m-fav-text'>FAVORITE</p>
          <div className='m-fav-btn'>
          </div>
        </NavLink>
    }

    return (
      <div className='m-movie'>
        <Link to={`/movie/${movie_id}`}>
          <div className='m-movie-poster'>
            <img
              className='m-poster'
              src={`http://image.tmdb.org/t/p/w342//${poster_path}`}
              alt='Click here for movie details.'/>
          </div>
        </Link>
        { favoriteBtn }
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  isLoggedIn: state.user.id,
})

export const mapDispatchToProps = (dispatch) => ({
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId)),
})

Movie.propTypes = {
  movie_id: PropTypes.number.isRequired,
  overview: PropTypes.string.isRequired,
  poster_path: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  vote_average: PropTypes.number,
  isFavorite: PropTypes.bool.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Movie);