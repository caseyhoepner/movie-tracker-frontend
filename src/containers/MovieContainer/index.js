import React from 'react';
import PropTypes from 'prop-types';
import './MovieContainer.css';
import Movie from '../../components/Movie';
import { connect } from 'react-redux';


export const MovieContainer = (props) => {
  const movies = props.movies.map(movie => {
    return <Movie { ...movie } key={movie.id} />
  })

  return (
    <div className='movie-container'>
      { movies }
    </div>
  )
}

const mapStateToProps = (state) => ({
  movies: state.movies
})

MovieContainer.propTypes = {
  movies: PropTypes.array.isRequired
}

export default connect(mapStateToProps)(MovieContainer);