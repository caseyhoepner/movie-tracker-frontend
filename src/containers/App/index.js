import React, { Component } from 'react';
import { Link, Route, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as API from '../../utils';
import apiKey from '../../API-key.js';
import { addMovies, toggleFavorite } from '../../actions';
import MovieContainer from '../MovieContainer';
import Nav from '../Nav';
import LoginForm from '../LoginForm'
import MovieDetails from '../../components/MovieDetails';
import wesIcon from '../../assets/images/wes.png';
import './App.css';

export class App extends Component {
  async componentDidMount() {
    const movies = await API.fetchMovies(apiKey)
    await this.props.addMovies(movies)
  }

  render() {
    const { user } = this.props
    return (
      <div className="a-App">
        <header className="a-header-container">
          <Link to='/' className="a-logo">
            <img
              src={ wesIcon }
              className="a-logo-icon"
              alt="An icon of Wes Anderson's face"
            />
            <h1 className='a-header-title'>WesTracker</h1>
          </Link>
          <Nav />
        </header>
        <Route
          exact path='/'
          component={ MovieContainer }
        />
        <Route
          exact path='/login'
          component={ LoginForm }
        />
        <Route path='/movie/:id' render={({match}) => {
          const { id } = match.params;
          const movie = this.props.movies.find(movie => (
            movie.id === parseInt(id, 10)))
          return (
            <MovieDetails {...movie} />
          )}} />
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  movies: state.movies
})

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies)),
  toggleFavorite: (movieId) => dispatch(toggleFavorite(movieId))
})

App.propTypes = {
  user: PropTypes.object.isRequired,
  addMovies: PropTypes.func.isRequired
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));