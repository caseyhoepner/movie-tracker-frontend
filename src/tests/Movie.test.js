import React from 'react';
import { shallow } from 'enzyme';
import { Movie, mapStateToProps, mapDispatchToProps } from '../containers/Movie';
import { mockFavMovies, mockUser, mockFavMovieId } from './testMocks';
import * as Actions from '../actions'

describe('Movie', () => {
  describe('Movie Component', () => {
    let wrapper;
    let mockMovie;

    beforeEach(() => {
      mockMovie = mockFavMovies[0]
      wrapper = shallow(<Movie
                          movie_id={mockMovie.movie_id}
                          overview={mockMovie.overview}
                          poster_path={mockMovie.poster_path}
                          release_date={mockMovie.release_date}
                          title={mockMovie.title}
                          vote_average={mockMovie.vote_average}
                          isFavorite={mockMovie.isFavorite}
                        />);
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the movies from state', () => {
      const mockState = {
        user: mockUser
      }

      const expected = mockUser.id

      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.isLoggedIn).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the toggleFavorites action', () => {
      const mockDispatch = jest.fn()
      const expected = Actions.toggleFavorite(mockFavMovieId)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(mockFavMovieId)

      expect(mockDispatch).toHaveBeenCalledWith(expected)
    })
  })
})