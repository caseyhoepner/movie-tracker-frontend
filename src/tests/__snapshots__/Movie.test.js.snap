// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Movie Component should match the snapshot 1`] = `
<div
  className="m-movie"
>
  <Link
    replace={false}
    to="/movie/1"
  >
    <div
      className="m-movie-poster"
    >
      <img
        alt="Click here for movie details."
        className="m-poster"
        src="http://image.tmdb.org/t/p/w342//poster-path1.jpg"
      />
    </div>
  </Link>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="m-fav-container"
    to="/login"
  >
    <p
      className="m-fav-text"
    >
      FAVORITE
    </p>
    <div
      className="m-fav-btn"
    />
  </NavLink>
</div>
`;
